src/Compilers/CSharp/Test/Emit3/Semantics/CollectionExpressionTests_WithElement_Constructor.cs

```diff
@@ -810,9 +809,9 @@ static void Main()
             """;
 
         CreateCompilation(source).VerifyDiagnostics(
-            // (20,28): error CS1729: 'MyList<int>' does not contain a constructor that takes 4 arguments
+            // (20,29): error CS1729: 'MyList<int>' does not contain a constructor that takes 4 arguments
             //         MyList<int> list = [with(name: "test", values: 1, 2, 3), 4];
-            Diagnostic(ErrorCode.ERR_BadCtorArgCount, @"[with(name: ""test"", values: 1, 2, 3), 4]").WithArguments("MyList<int>", "4").WithLocation(20, 28));
+            Diagnostic(ErrorCode.ERR_BadCtorArgCount, @"with(name: ""test"", values: 1, 2, 3)").WithArguments("MyList<int>", "4").WithLocation(20, 29));
```

Comment:
the syntax location was improved, so a lot of diagnostics related to 'with(...)' element are now just placed on that.